cmd/uplink/cmd/setup.go

```diff
@@ -4,48 +4,41 @@
 package cmd
 
 import (
-	"crypto/rand"
 	"fmt"
 	"os"
 	"path/filepath"
 
-	base58 "github.com/jbenet/go-base58"
 	"github.com/spf13/cobra"
 	"go.uber.org/zap"
 
 	"storj.io/storj/internal/fpath"
 	"storj.io/storj/pkg/cfgstruct"
 	"storj.io/storj/pkg/miniogw"
 	"storj.io/storj/pkg/process"
-	"storj.io/storj/pkg/provider"
 )
 
+// UplinkFlags configuration flags
+type UplinkFlags struct {
+	APIKey        string `default:"" help:"the api key to use for the satellite" setup:"true"`
+	SatelliteAddr string `default:"localhost:7778" help:"the address to use for the satellite" setup:"true"`
+
+	miniogw.Config // TODO remove when GetMetainfo will be moved from minio
+}
+
 var (
 	setupCmd = &cobra.Command{
 		Use:         "setup",
 		Short:       "Create an uplink config file",
 		RunE:        cmdSetup,
 		Annotations: map[string]string{"type": "setup"},
 	}
-	setupCfg struct {
-		Identity           provider.IdentitySetupConfig
-		APIKey             string `default:"" help:"the api key to use for the satellite" setup:"true"`
-		EncKey             string `default:"" help:"your root encryption key" setup:"true"`
-		GenerateMinioCerts bool   `default:"false" help:"generate sample TLS certs for Minio GW" setup:"true"`
-		SatelliteAddr      string `default:"localhost:7778" help:"the address to use for the satellite" setup:"true"`
-
-		Server miniogw.ServerConfig
-		Minio  miniogw.MinioConfig
-		Client miniogw.ClientConfig
-		RS     miniogw.RSConfig
-		Enc    miniogw.EncryptionConfig
-	}
+
+	setupCfg UplinkFlags
 
 	defaultConfDir     = fpath.ApplicationDir("storj", "uplink")
 	defaultIdentityDir = fpath.ApplicationDir("storj", "identity", "uplink")
 
 	cliConfDir  string
```

Comment:
be consistent with other binary implementations

Reply:
done